packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Models/Types/EnumTypeProvider.cs

```diff
@@ -0,0 +1,386 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Globalization;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.Generator.CSharp.Expressions;
+using Microsoft.Generator.CSharp.Input;
+using static Microsoft.Generator.CSharp.Expressions.Snippets;
+
+namespace Microsoft.Generator.CSharp
+{
+    public class EnumTypeProvider : TypeProvider
+    {
+        private readonly IReadOnlyList<InputEnumTypeValue> _allowedValues;
+        private readonly ModelTypeMapping? _typeMapping;
+
+        public EnumTypeProvider(InputEnumType input, SourceInputModel? sourceInputModel) : base(sourceInputModel)
+        {
+            _allowedValues = input.AllowedValues;
+            _deprecated = input.Deprecated;
+
+            if (input.Accessibility == "internal")
+            {
+                DeclarationModifiers = TypeSignatureModifiers.Internal;
+            }
+            else
+            {
+                DeclarationModifiers = TypeSignatureModifiers.Public;
+            }
+            IsAccessibilityOverridden = input.Accessibility != null;
```

Comment:
do we need this property at all?  It doesn't appear to be used anywhere

Reply:
yeah this is a shimming layer for our post processor and the accessibility override coming from TCGC.
We do not have anything related with it yet and we do not have to have it now

Reply:
I think we leave this off for now as post processing will be getting a big overhaul in that we will adjust the TypeProvider based on existing / custom code before passing to the writer so to reduce / eliminate post processing which is very costly.